### PR TITLE
[PodSecurity]Add context to failure message

### LINKED ISSUE #103561: [PodSecurity] Improve error UX
Ideas to consider:
- [x] ~~Group errors by container. Ex:
    `pod: check1, check2 (value); container1: check1 (value, value); init container2: check1, check5~~
- [x] ~~Consider making reasons more machine-readable~~
- [x] https://github.com/kubernetes/kubernetes/issues/103213
- [x] Add context to failure message, like `Pod violates baseline PodSecurity level:` - https://github.com/kubernetes/kubernetes/pull/105314
- [x] Fix invalid namespace label error message - https://github.com/kubernetes/kubernetes/pull/105959

Hoisted to followup in #106129

### PR DESCRIPTION
**Pull request by calvin0327**
#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:
Provide more context for the failure message.

#### Which issue(s) this PR fixes:

For #103561

#### Special notes for your reviewer:
I worked on it for a few days, I don't know that is reasonable, correct me if not.

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by tallclair — COMMENTED** [review]
Thanks for picking this up!

File: staging/src/k8s.io/pod-security-admission/admission/admission.go [hunk 1]
```diff
@@ -412,13 +412,23 @@ func (a *Admission) EvaluatePod(ctx context.Context, nsPolicy api.Policy, nsPoli
...
 		}
 	}
 
 	// TODO: reuse previous evaluation if audit level+version is the same as enforce level+version
 	if result := policy.AggregateCheckResults(a.Evaluator.EvaluatePod(nsPolicy.Audit, podMetadata, podSpec)); !result.Allowed {
-		auditAnnotations["audit"] = result.ForbiddenDetail()
+		auditAnnotations["audit"] = fmt.Sprintf(
+			"Pod violates %q version of %q PodSecurity profile: %s",
+			nsPolicy.Warn.Version.String(),
+			nsPolicy.Warn.Level,
```
[the remark this answers is not in the document]
Comment (calvin0327) [reply staging/src/k8s.io/pod-security-admission/admission/admission.go:429]: Thanks for suggestion.

**Review by tallclair — COMMENTED** [review]
last round of comments 🙂

**Review by tallclair — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**